apps/core/scenes/results/ResultsListItem.js

```diff
@@ -1,61 +1,246 @@
+    const { route } = this.props.data;
+    if (route) {
+      return route.map((routeItem: RouteItem) =>
+        this.getSanitizedRouteItem(routeItem),
+      );
+    }
+    return [];
+  };
+
+  getSector = (
+    route: Array<TripSectorWithId>, // rename to sanitizeRoute
+    demandedSector: number,
+    sectorBorderIndex: ?number,
+  ): Array<TripSectorWithId> => {
+    if (
+      demandedSector === null ||
+      sectorBorderIndex === null ||
+      sectorBorderIndex === -1
+    ) {
+      return route;
+    }
+    if (demandedSector === 0) {
+      return route.slice(0, sectorBorderIndex);
+    }
+    return route.slice(sectorBorderIndex);
+  };
+
+  getDurationBetweenSectors = (route: ?Route, sectorBorderIndex: ?number) => {
+    if (route && sectorBorderIndex != null && sectorBorderIndex > 0) {
+      const stopBeforeSector = route[sectorBorderIndex - 1];
+      const stopAfterSector = route[sectorBorderIndex];
+
+      const utcDeparture =
+        stopAfterSector &&
+        stopAfterSector.departure &&
+        DateTime.fromISO(stopAfterSector.departure.utcTime);
+      const utcArrival =
+        stopBeforeSector &&
+        stopBeforeSector.arrival &&
+        DateTime.fromISO(stopBeforeSector.arrival.utcTime);
+      return this.getDateTimeInterval(utcDeparture, utcArrival, [
+        'days',
+        'hours',
+      ]);
+    }
     return null;
-  }
-  const { flyFrom, flyTo, localDeparture, localArrival, price } = props.data;
-  /**
-   * TODO: properly handle possible undefined props in final version of list
-   * (for example: should be entry with undefined price still displayed?)
-   *
-   * NOTE: ugly form of string creation below is used for now because
-   * using ?? inside template literals caused problems with code styling
-   * inside VS Code (this part will be probably completely replaced
-   * in next iteration, so it's temporary solution)
-   */
-  const itemString =
-    (flyFrom ?? 'Unknown destination') +
-    ' / ' +
-    (localDeparture ?? 'Unknown departure') +
-    ' -> ' +
-    (flyTo ?? 'Unknown destination') +
-    ' / ' +
-    (localArrival ?? 'Unknown arrival') +
-    ' | ' +
-    (price ?? 'Unknown price');
-  return <Text style={styles.listItem}>{itemString}</Text>;
-};
+  };
 
-const styles = StyleSheet.create({
-  listItem: {
-    fontSize: 12,
-    padding: 2,
-    margin: 2,
-    backgroundColor: defaultTokens.backgroundBody,
-  },
-});
+  render() {
+    const { data } = this.props;
+    const sanitizedRoute = this.getSanitizedRoute();
+    const sectorBorderIndex = this.getSectorBorderIndex();
+    const hasSectorBorder = sectorBorderIndex != null && sectorBorderIndex > -1;
+    if (data == null) {
+      return null;
+    }
+    const { price } = data;
+    /**
+     * TODO: properly handle possible undefined props in final version of list
+     * (for example: should be entry with undefined price still displayed?)
+     *
+     * NOTE: ugly form of string creation below is used for now because
```

Comment:
This NOTE is probably no longer relevant and can be removed.

Reply:
OK